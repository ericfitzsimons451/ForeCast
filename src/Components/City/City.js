import React from 'react'
import '../../SCSS/Main.scss'
import { Link } from 'react-router-dom'
import { PropTypes } from 'prop-types'

export const City = ({ currentWeather }) => {
	if (currentWeather.weather) {
		return (
			<div className='card'>
				<h2>{currentWeather.city_name}</h2>
				<h3>Temperature: {((currentWeather.temp * 9) / 5) + 32} F</h3>
				<h3>Relative Humidity: {currentWeather.app_temp} %</h3>
				<h3>Description: {currentWeather.weather.description}</h3>
				<h3>Cloud Coverage: {currentWeather.clouds}%</h3>
				<h3>Sunrise: {currentWeather.sunrise}</h3>
				<h3>Sunset: {currentWeather.sunset}</h3>
				<Link to='/search'>Return</Link>
			</div>
		)
	} else {
		return (
			<div>Loading, please wait...</div>
		)
	}
}

City.propTypes = {
	currentWeather: PropTypes.object
}

export default City
