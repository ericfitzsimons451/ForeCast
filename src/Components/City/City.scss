.card {
    border: solid 1px red;
    border-radius: 10px;
    width: 32vw;
    height: 30vh;
    margin: 10px;
    &:hover {
        box-shadow: 2px 2px;
        cursor: pointer
    }
}

.city-name {
    font-size: 2rem;
}