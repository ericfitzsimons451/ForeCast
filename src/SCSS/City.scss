.card {
	background: rgba(169, 176, 168, 0.815);
	width: 50vw;
	height: 55vh;
	margin-top: 30px;
	border-radius: 15px;
	display:flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	h3 {
		font-size: 1.4rem;
		margin: 10px;
	}
	a {
		&:hover {
			cursor: pointer;
		}
	}
}

.name {
	font-size: 2rem;
	margin: 10px;
}

.link-return {
	border: 1px solid black;
	border-radius: 15px;
	width: 30vw;
	height: 30px;
	font-size: 1.8rem;
	padding: 20px;
	text-decoration: none;
}

@media only screen and (max-width: 615px) {
	.card {
		margin-top: 5px;
		h3 {
			font-size: 1.1rem;
			margin: 8px;
		}
	}
	
	.link-return {
		font-size: 1.5rem;
	}
}

@media only screen and (max-width: 490px) {
	.card {
		h3 {
			font-size: 1rem;
		}
	}
}	

