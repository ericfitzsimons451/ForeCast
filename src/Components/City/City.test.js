import React from 'react'
import { City } from './City'
import { shallow } from 'enzyme'


describe('City', () => {
    it('should match the snapshot if there is no data', () => {
        const weather = {}
        let wrapper = shallow(<City weather={weather} />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot if data is passed down', () => {
        const weather = {
            city: 'Denver'
        }
        let wrapper = shallow(<City weather={weather} />)
        expect(wrapper).toMatchSnapshot()
    })
})