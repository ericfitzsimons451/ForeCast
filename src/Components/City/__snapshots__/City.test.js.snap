// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City should match the snapshot if data is passed down 1`] = `
<div>
  Loading, please wait...
</div>
`;

exports[`City should match the snapshot if there is no data 1`] = `
<div>
  Loading, please wait...
</div>
`;
